### PR TITLE
refactor(components): improve code

### PR DESCRIPTION
**Pull request by sxzz**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/calendar/src/calendar.ts [hunk 1]
```diff
@@ -10,23 +10,25 @@ export type CalendarDateType =
...
-      range.every((item) => item instanceof Date),
+    validator: isValidRange,
   },
 } as const)
 export type CalendarProps = ExtractPropTypes<typeof calendarProps>
 
 export const calendarEmits = {
   [UPDATE_MODEL_EVENT]: (value: Date) => value instanceof Date,
-  input: (value: Date) => value instanceof Date,
+  [INPUT_EVENT]: (value: Date) => value instanceof Date,
```
Comment (btea) [inline packages/components/calendar/src/calendar.ts:31]:
```suggestion
  [INPUT_EVENT]: (value: Date) => isDate(value),
```

File: packages/components/calendar/src/calendar.ts [hunk 2]
```diff
@@ -10,23 +10,25 @@ export type CalendarDateType =
   | 'next-year'
   | 'today'
 
+const isValidRange = (range: unknown): range is [Date, Date] =>
+  Array.isArray(range) &&
```
Comment (btea) [inline packages/components/calendar/src/calendar.ts:14]:
```suggestion
  isArray(range) &&
```

File: packages/components/calendar/src/calendar.ts [hunk 3]
```diff
@@ -10,23 +10,25 @@ export type CalendarDateType =
...
+  range.length === 2 &&
+  range.every((item) => item instanceof Date)
```
Comment (btea) [inline packages/components/calendar/src/calendar.ts:16]:
```suggestion
  range.every((item) => isDate(item))
```

**Comment by sxzz** [comment]
@btea Updated.

**Comment by btea** [comment]
> @btea Updated.

👌

**Review by btea — APPROVED** [review]

**Pull request merged**